docs/user-manual/xr/ar/depth-sensing.md

```diff
@@ -0,0 +1,176 @@
+
+Depth sensing - provides access to depth estimation of real-world objects in real-time. Underlying systems might have different methods of estimations such as Lidar hardware or Computer Vision, which provide various levels of quality and reliability.
+
+WebXR Depth Sensing provides access to depth information for each view and matches the color information. Various browsers might implement two paths: CPU and GPU, with various performance impacts depending on the path. PlayCanvas integrates an API abstracting away as much as possible the difference, e.g. texture is available for CPU and GPU paths.
+
+Platforms might implement either path CPU or GPU, or even both.
+
+To request access to camera depth, the session should be started with an extra flag:
+
+```javascript
+app.xr.start(camera, pc.XRTYPE_AR, pc.XRSPACE_LOCALFLOOR, {
+    depthSensing: { // request access to camera depth
+        usagePreference: pc.XRDEPTHSENSINGUSAGE_GPU, // prefer GPU implementation
+        dataFormatPreference: pc.XRDEPTHSENSINGFORMAT_F32 // prefer data as Float 32 array/texture
+    }
+});
+```
+
+## Support
+
+You can check if the camera depth is supported by the system:
+
+```javascript
+if (app.xr.views.supportedDepth) {
+    // camera depth access is supported
+}
+
+app.xr.on('start', () => {
+    if (app.xr.views.availableDetpth) {
+        // camera depth information is available
+
+        if (app.xr.views.depthGpuOptimized) {
+            // GPU path
+        } else {
+            // CPU path
+        }
+    }
+});
+```
+
+## Distance Measurements
+
+Depth estimation and availability of the data a subject to underlying AR system reliability, so depth information might not be always available.
+
+WebXR supports it only for CPU-path. Using Depth Sensing, you can measure the distance by providing U and V, which are 0 to 1 coordinates of a screen (left-right top-bottom).
+
+```javascript
+// get monoscope view (mobile screens)
+const view = app.xr.views.get(pc.XREYE_NONE);
+if (view) {
+    // get distance from the middle of a screen
+    const distance = view.getDepth(0.5, 0.5);
+
+    if (distance !== null) {
+        // distance is in meeters
+    }
+}
+```
+
+## Texture
+
+You can access a texture of the depth. PlayCanvas augments the different CPU/GPU paths and provides one texture that can be an array texture in the case of stereoscopic screens (e.g. HMDs).
+
+Accessing the texture:
+
+```javascript
+const view = app.xr.views.list[0];
+if (view) {
+    const texture = view.textureDepth;
+
+    if (texture) {
+        // get global uniform
+        const scopeDepthMap = app.graphicsDevice.scope.resolve('depthMap');
+        // set uniform
+        scopeDepthMap.setValue(texture);
+    }
+}
+```
+
+### Stereo Views
+
+When using texture in the shader, depending on monoscope or stereoscope scenario, a different approach should be used. And can be implemented by `#define` in the shader:
```

Comment:
```suggestion
When using the depth texture in the shader, depending on a monoscope or stereoscope scenario, a different approach should be used. This can be implemented by a `#define` in the shader:
```